### ISSUE TITLE
500 error from generateCaptcha() function

### ISSUE THREAD

**Issue by physiii** [userfrosting/UserFrosting#484]
Can't login or register after setting up master account, error.log:

[Sat Jan 02 15:56:34.155051 2016] [:error] [pid 2145] [client 192.168.0.3:58434] PHP Fatal error:  Call to undefined function UserFrosting\imagecreatetruecolor() in /var/www/html/assets/controllers/BaseController.php on line 118
[Sat Jan 02 15:56:34.163277 2016] [:error] [pid 2145] [client 192.168.0.3:58434] Fatal error (1) in /var/www/html/assets/controllers/BaseController.php on line 118: Call to undefined function UserFrosting\imagecreatetruecolor()
[Sat Jan 02 15:59:06.609999 2016] [:error] [pid 2147] [client 192.168.0.3:58524] PHP Fatal error:  Call to undefined function UserFrosting\imagecreatetruecolor() in /var/www/html/assets/controllers/BaseController.php on line 118, referer: http://192.168.0.31/account/login
[Sat Jan 02 15:59:06.611898 2016] [:error] [pid 2147] [client 192.168.0.3:58524] Fatal error (1) in /var/www/html/assets/controllers/BaseController.php on line 118: Call to undefined function UserFrosting\imagecreatetruecolor(), referer: http://192.168.0.31/account/login

**Comment by frostbitten** [comment]
From [stack overflow](http://stackoverflow.com/questions/4560996/call-to-undefined-function-imagecreatetruecolor-error-in-php-pchart):

Use the following code to test if you have GD extension:

```
$testGD = get_extension_funcs("gd"); // Grab function list 
if (!$testGD){ echo "GD not even installed."; exit; }
echo "<pre>".print_r($testGD,true)."</pre>";
```

**Comment by physiii** [comment]
Great, I just had to run apt-get install php5-gd and it works!

Thank you!

**Comment by alexweissman** [comment]
Weird, the installer should be detecting this condition: https://github.com/userfrosting/UserFrosting/blob/master/userfrosting/controllers/InstallController.php#L114-L120

This is also covered in the troubleshooting guide, maybe we can clarify it better?  http://www.userfrosting.com/troubleshooting/#gd-error

**Comment by alexweissman** [comment]
From the [PHP documentation](http://php.net/manual/en/function.imagecreatetruecolor.php): 

> Depending on your PHP and GD versions this function is defined or not. With PHP 4.0.6 through 4.1.x this function always exists if the GD module is loaded, but calling it without GD2 being installed PHP will issue a fatal error and exit. With PHP 4.2.x this behaviour is different in issuing a warning instead of an error. Other versions only define this function, if the correct GD version is installed.

uhhh...what?

**Comment by Netrilix** [comment]
http://php.net/manual/en/function.imagecreatetruecolor.php#28470

That might help us.

**Comment by alexweissman** [comment]
Yeesh.  But we should only require `gd`, not `gd2`, right?

**Comment by Netrilix** [comment]
We're only using it for CAPTCHA, correct? We should be able to get by with just `GD`, but that'll mean using `imagecreate()` instead of `imagecreatetruecolor()`. If we wanted to support `imagecreatetruecolor()` we'd need to do the ugly version detection to make sure `GD2` is present or fall back to `GD`-only functions.

**Comment by alexweissman** [comment]
The [`CheckEnvironment` class](https://github.com/userfrosting/UserFrosting/blob/b48f29eae3a62b235a00f982dbf9f80fd1c03c23/app/sprinkles/core/src/Util/CheckEnvironment.php#L156-L194) in `dev` checks for all of the required `image*` functions now, so this should be solved for 4.0.
